awx/ui_next/src/App.jsx

```diff
@@ -1,216 +1,89 @@
-      });
-    } catch (err) {
-      this.setState({ configError: err });
-    }
-  }
-
-  render() {
-    const {
-      ansible_version,
-      custom_virtualenvs,
-      project_base_dir,
-      project_local_paths,
-      isAboutModalOpen,
-      isNavOpen,
-      me,
-      version,
-      configError,
-    } = this.state;
-    const { i18n, routeConfig = [], navLabel = '', children } = this.props;
-
-    const header = (
-      <PageHeader
-        showNavToggle
-        onNavToggle={this.handleNavToggle}
-        logo={<BrandLogo />}
-        logoProps={{ href: '/' }}
-        toolbar={
-          <PageHeaderToolbar
-            loggedInUser={me}
-            isAboutDisabled={!version}
-            onAboutClick={this.handleAboutOpen}
-            onLogoutClick={this.handleLogout}
-          />
-        }
-      />
-    );
-
-    const sidebar = (
-      <PageSidebar
-        isNavOpen={isNavOpen}
-        theme="dark"
-        nav={
-          <Nav aria-label={navLabel} theme="dark">
-            <NavList>
-              {routeConfig.map(({ groupId, groupTitle, routes }) => (
-                <NavExpandableGroup
-                  key={groupId}
-                  groupId={groupId}
-                  groupTitle={groupTitle}
-                  routes={routes}
-                />
-              ))}
-            </NavList>
-          </Nav>
-        }
-      />
-    );
-
-    return (
-      <Fragment>
-        <Page usecondensed="True" header={header} sidebar={sidebar}>
-          <ConfigProvider
-            value={{
-              ansible_version,
-              custom_virtualenvs,
-              project_base_dir,
-              project_local_paths,
-              me,
-              version,
-            }}
-          >
-            {children}
-          </ConfigProvider>
-        </Page>
-        <About
-          ansible_version={ansible_version}
-          version={version}
-          isOpen={isAboutModalOpen}
-          onClose={this.handleAboutClose}
-        />
-        <AlertModal
-          isOpen={configError}
-          variant="error"
-          title={i18n._(t`Error!`)}
-          onClose={this.handleConfigErrorClose}
-        >
-          {i18n._(t`Failed to retrieve configuration.`)}
-          <ErrorDetail error={configError} />
-        </AlertModal>
-      </Fragment>
-    );
-  }
+  useRouteMatch,
+  useLocation,
+  HashRouter,
+  Route,
+  Switch,
+  Redirect,
+} from 'react-router-dom';
+import { I18n, I18nProvider } from '@lingui/react';
+
+import AppContainer from './components/AppContainer';
+import Background from './components/Background';
+import NotFound from './screens/NotFound';
+import Login from './screens/Login';
+
+import ja from './locales/ja/messages';
+import en from './locales/en/messages';
+import { isAuthenticated } from './util/auth';
+import { getLanguageWithoutRegionCode } from './util/language';
+
+import getRouteConfig from './routeConfig';
+
+const AppRoute = ({ auth, children, ...rest }) =>
```

Comment:
This function is confusing, but you and I were able to clear it up offline.

Reply:
You're right! Should be fixed now.